fishtank/src/cluster.ts

```diff
@@ -87,6 +87,36 @@ export class Cluster {
     return stdout
   }
 
+  async runNodeCommand(options: {
+    name: string
+    args: string[]
+    image?: string
+  }): Promise<string> {
```

Comment:
This method is meant to act on a node, but it's in the `Cluster` class instead of the `Node` class. I'm also a bit concerned about this method creating directories but not offering any way to do any cleanup.

If I had to guess, this would be the workflow that you want to accomplish:

```ts
const node = cluster.spawn({ image: 'old-image' }); // start a new node
node.stop(); // this needs to be implemented
cluster.runNodeCommand({ name: node.name, args: ["migrations:start"], image: 'new-image' });
cluster.spawn({ name: node.name, image: 'new-image' });
// ...verify that the node is working and using the latest schema
```

but how about this instead:

```ts
const node = cluster.spawn({ image: 'old-image' }); // start a new node
node.stop(); // this needs to be implemented
node.respawn({ cmd: ["migrations:start"] }); // this could be a wrapper on cluster.spawn() or internalSpawn() that automatically injects the right name, and maybe tells it not to touch any config files
node.respawn({ name: node.name, image: 'new-image' });
// ...verify that the node is working and using the latest schema
```

Would this fit your use case?

To implement this, you'd need to expand `spawn()` and/or `internalSpawn()` to accept a new `cmd` argument to override the default `['start']` command, and without the need to duplicate code.

Reply:
Yes that's the workflow that I'm looking for. I'll rewrite this to allow commands to operate on the node rather than the cluster